src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -504,5 +515,24 @@ private static string[] SplitFlagsEnum(string value)
 #endif
                 );
         }
+
+        private static bool ContainsSpecialChar()
+        {
+#if NETCOREAPP
+            string[] names = Enum.GetNames<T>();
```

Comment:
We're now calling `Enum.GetNames` twice, would recommend removing this method and inlining the foreach check in the constructor.

Reply:
make sense.

Reply:
Correct me if I'm wrong.
I think `s_specialChars` is a static field, it is initialized only once when invoke constructor of EnumConverter<T> for the first time.

Reply:
You're correct, however I don't think storing a static boolean is an optimization that is worth it. Converters are only meant to be initialized once within an application's lifetime. Moreover this approach makes it difficult to report back to the user what particular identifier contains the unsupported characters.